x-pack/plugins/search_inference_endpoints/public/components/inference_flyout_wrapper_component.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexItem, EuiCallOut, EuiText, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { InferenceFlyoutWrapper } from '@kbn/inference_integration_flyout/components/inference_flyout_wrapper';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { InferenceTaskType } from '@elastic/elasticsearch/lib/api/types';
+
+import { i18n } from '@kbn/i18n';
+import { ModelConfig } from '@kbn/inference_integration_flyout/types';
+import { extractErrorProperties } from '@kbn/ml-error-utils';
+import { TrainedModelConfigResponse } from '@kbn/ml-plugin/common/types/trained_models';
+import { SUPPORTED_PYTORCH_TASKS, TRAINED_MODEL_TYPE } from '@kbn/ml-trained-models-utils';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { INFERENCE_ENDPOINTS_QUERY_KEY } from '../../common/constants';
+import { useKibana } from '../hooks/use_kibana';
+import { docLinks } from '../../common/doc_links';
+
+interface InferenceFlyoutWrapperComponentProps {
+  isInferenceFlyoutVisible: boolean;
+  setIsInferenceFlyoutVisible: (isVisible: boolean) => void;
+}
+
+export const InferenceFlyoutWrapperComponent: React.FC<InferenceFlyoutWrapperComponentProps> = ({
+  isInferenceFlyoutVisible,
+  setIsInferenceFlyoutVisible,
+}) => {
+  const [inferenceAddError, setInferenceAddError] = useState<string | undefined>(undefined);
+  const [isCreateInferenceApiLoading, setIsCreateInferenceApiLoading] = useState(false);
+  const [availableTrainedModels, setAvailableTrainedModels] = useState<
+    TrainedModelConfigResponse[]
+  >([]);
+
+  const [inferenceEndpointError, setInferenceEndpointError] = useState<string | undefined>(
+    undefined
+  );
+
+  const queryClient = useQueryClient();
+
+  const {
+    services: { ml },
+  } = useKibana();
+
+  const createInferenceEndpointMutation = useMutation(
+    async ({
+      inferenceId,
+      taskType,
+      modelConfig,
+    }: {
+      inferenceId: string;
+      taskType: InferenceTaskType;
+      modelConfig: ModelConfig;
+    }) => {
+      await ml?.mlApi?.inferenceModels?.createInferenceEndpoint(inferenceId, taskType, modelConfig);
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([INFERENCE_ENDPOINTS_QUERY_KEY]);
+      },
+    }
+  );
+
+  const onInferenceEndpointChange = async () => {};
```

Comment:
are we planning to do something with this later?

Reply:
Actually, we should utilize this callback to display error if the inference_endpoint already exists. I have modified this part.